File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Binaries/Multiply.cs
using AST.Nodes.Interfaces;


namespace AST.Nodes.Expressions.Binaries
{
    public class Multiply : Binary
    {
        public Multiply(Expression left, Expression right, PositionInText position)
            : base(left, right, position)
        { }

        public override T Accept<T>(VisitorInterfaces.IExpressionVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "*";
        }
    }
}

File: KevinRene/src/ql/ast/expression/arithmetic/Multiply.java
package ql.ast.expression.arithmetic;

import ql.ast.Expression;
import ql.ast.QLType;
import ql.ast.expression.Binary;
import ql.ast.type.QLNumeric;
import ql.ast.visitor.ExpressionVisitor;

public class Multiply extends Binary {
	public Multiply(Expression left, Expression right) {
		super(left, right, "*");
	}
	
	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {		
		return visitor.visit(this);
	}
	
	@Override
	public QLType getType() {
		return new QLNumeric();
	}
}
